### PR TITLE
Bug 1185269 - [Settings] Convert radio controls to use gaia-radio

### PR DESCRIPTION
**Pull request by KevinGrandon**
https://bugzilla.mozilla.org/show_bug.cgi?id=1185269

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=55d3f3d735dfd8fdbcbc4cde86493eab9dd4967c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0fa7195458d745a9f7ca994b8da11f4080057455

File: apps/settings/elements/do_not_track.html [hunk 1]
```diff
@@ -8,42 +8,30 @@ <h1 data-l10n-id="doNotTrack-header"></h1>
...
         </li>
         <li class="description">
-          <label class="pack-radio">
-            <input type="radio" name="privacy.donottrackheader.value" value="0">
-            <span data-l10n-id="allowTracking">Tell websites and apps that I want to be tracked</span>
+          <gaia-radio name="privacy.donottrackheader.value" value="0">
+            <label data-l10n-id="allowTracking"></label>
           </label>
-        </li>
+        </ligaia-radio
```
Comment (timvandermeij) [inline apps/settings/elements/do_not_track.html]: There appears to be a typo here.

Comment (KevinGrandon) [inline apps/settings/elements/do_not_track.html]: thanks :)

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=115909056724b1be688cb491ef940b568a6b7fb3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1e4de9d5bacb5842cc53ddd13669d97b2a9c9960

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cf747b83180d78f62a14ffdb8355bb74b87e1d9e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5da9ca92d0bdfbfcdbcb316d831dcaa78f38baeb

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ddc736b38cd7cd8fc09d386f2702f8886d183fc1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5510d458e1baf3a31e1a436840b58963086deaf4

File: apps/settings/js/modules/panel_utils.js [hunk 2]
```diff
@@ -30,10 +30,10 @@ define(function(require) {
 
       request.onsuccess = function() {
         switch (input.type) {
-          case 'radio':
+          case 'gaia-radio':
             input.checked = (input.value == request.result[key]);
             break;
-          case 'checkbox':
+          case 'gaia-checkbox':
```
Comment (gasolin) [inline apps/settings/js/modules/panel_utils.js]: Have we removed all checkbox? if not we can't just change the cases yet. Please add another gaia-checkbox case instead

File: apps/settings/js/modules/panel_utils.js [hunk 3]
```diff
@@ -82,12 +82,12 @@ define(function(require) {
         fields.forEach(function(input) {
           key = input.dataset.setting;
           switch (input.type) {
-            case 'radio':
+            case 'gaia-radio':
```
Comment (gasolin) [inline apps/settings/js/modules/panel_utils.js]: same here

File: apps/settings/js/modules/panel_utils.js [hunk 4]
```diff
@@ -82,12 +82,12 @@ define(function(require) {
...
               if (input.checked) {
                 cset[key] = input.value;
               }
               break;
-            case 'checkbox':
+            case 'gaia-checkbox':
```
Comment (gasolin) [inline apps/settings/js/modules/panel_utils.js]: same here

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=87ab589144d163cf1f0a2cac90d693ec430e32ce

File: apps/settings/js/modules/panel_utils.js (hunk 3 above)
Comment (KevinGrandon) [inline apps/settings/js/modules/panel_utils.js]: Settings should no longer use any `input type="radio"` elements as we should be using gaia-radio everywhere now.

Comment (gasolin) [inline apps/settings/js/modules/panel_utils.js]: :+1:

File: apps/settings/js/modules/panel_utils.js [hunk 5]
```diff
@@ -383,8 +385,11 @@ define(function(require) {
       }
 
       // update <input> values when the corresponding setting is changed
-      var inputs = [].slice.call(panel.querySelectorAll(`input[name="${key}"],
-        gaia-switch[name="${key}"], gaia-checkbox[name="${key}"]`));
+      var inputs = [].slice.call(panel.querySelectorAll(
+        `input[name="${key}"],
+        gaia-switch[name="${key}"],
+        gaia-checkbox[name="${key}"],
+        gaia-checkbox[name="${key}"]`));
```
Comment (gasolin) [inline apps/settings/js/modules/panel_utils.js]: a dup?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=15e44de990cb2d63960a206c747681fc987aa550

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=84ee3630ff1b0be2d896f69462b9301147c59b03

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2850671fd6ed578ff1dadbeeab06327fea55fb42

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cffb0c75e995cc3f5e47ead7f6d4d837dcc8ce0f

**Pull request merged**